pkg/dataframe/series.go

```diff
@@ -215,10 +215,14 @@ func (a *seriesAggregator) getLabelNames() []string {
 		ls  labels.Labels
 		ret []string
 	)
+	// Create a union set (map) of all the label names.
+	lsMap := make(map[string]string)
```

Comment:
Any unit test to show what exactly this fixes? (:

Reply:
Should I create a test for the `series` package?

Reply:
Would be nice, but it's fine to do in next PR. Test would be really important to ensure this code does what we want ;p